File: demo/common/src/main/java/com/serious/demo/TestService.java
package com.serious.demo;
/*
 * @COPYRIGHT (C) 2023 Andreas Ernst
 *
 * All rights reserved
 */

import com.serious.service.Service;
import com.serious.service.ServiceInterface;

/**
 * @author Andreas Ernst
 */
@ServiceInterface(name = "test service")
public interface TestService extends Service {
    String hello();
}

File: demo/common/src/main/java/com/serious/demo/TestService.kt
package com.serious.demo
/*
* @COPYRIGHT (C) 2023 Andreas Ernst
*
* All rights reserved
*/

import com.serious.service.Service
import com.serious.service.ServiceInterface

@ServiceInterface(name = "test service")
interface TestService : Service {
    fun hello(): String

    fun throwException() : Void
}
